File: arc094_b_worst_case.rb
gets.to_i.times{a,b=gets.split(' ').map(&:to_i);r=Math.sqrt(a*b).to_i;n=!(r*(r+1)<a*b);p 2*(r-(n ?1:0))-(!((n&&r*r<a*b)||a==b)?1:0)}

File: arc094_b_worst_case.py
import math
for A,B in[map(int,input().split())for _ in range(int(input()))]:
 R=int(math.sqrt(A*B))
 N=not R*(R+1)<A*B
 print(2*(R-N)-(not ((N and R*R<A*B) or A==B)))
